aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -72,27 +49,34 @@ Tensor prepare_matrix_for_cublas(Tensor& tensor, bool& transpose_tensor) {
   return tensor_;
 }
 
+namespace {
+
 Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& mat1, const Tensor& mat2, Scalar beta, Scalar alpha) {
-  TORCH_CHECK(
```

Comment:
why don't we need to check for both mat1 and self dim size anymore?

Reply:
This check has been split into two. In the new version, first we check mat1 and mat2, after that we broadcast self (if applicable) and resize the result (if applicable), after that we do the check for self and result.